### PR TITLE
feat!: expose global master access in GA modules

### PR DESCRIPTION
**Pull request by ericyz**
The Global Master Access is already in [GA TF provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#nested_master_global_access_config).

This change will expose private_cluster_config.master_global_access_config in available GA modules

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/private-cluster/variables.tf [hunk 1]
```diff
@@ -366,6 +366,13 @@ variable "master_ipv4_cidr_block" {
   default     = "10.0.0.0/28"
 }
 
+variable "master_global_access_enabled" {
+  type        = bool
+  description = "Whether the cluster master is accessible globally (from any region) or only within the same region as the private endpoint."
+
+  default = true
```
Comment (bharathkkb) [inline modules/private-cluster/variables.tf:373]: @ericyz can we default this to false for now and open an issue to flip this to true in a following breaking release. This will allow us to put this in a minor release for now.

Comment (ericyz) [reply modules/private-cluster/variables.tf:373]: Sure.

**Comment by ericyz** [comment]
Updated. @bharathkkb - please review

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by ericyz** [comment]
@bharathkkb - I think the test case was failed due to other reasons. Could you please help to verify and let me know any chance required for this MR

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by ericyz** [comment]
Hi @bharathkkb , any chance to review and provide feedback? This MR is on the critical path on the project. (rebased to master)

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

File: autogen/main/variables.tf.tmpl [hunk 2]
```diff
@@ -402,24 +402,26 @@ variable "enable_private_endpoint" {
 variable "enable_private_nodes" {
   type        = bool
   description = "(Beta) Whether nodes have internal IP addresses only"
-  default     = false
+  default     = true
```
Comment (bharathkkb) [inline autogen/main/variables.tf.tmpl:405]: @ericyz This seems unrelated?

**Review by bharathkkb — CHANGES_REQUESTED** [review]
@ericyz Sorry for the delay, CI had some issues. You could remove that unrelated change from the PR and this should be good to go

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main/variables.tf.tmpl (hunk 2 above)
Comment (ericyz) [reply autogen/main/variables.tf.tmpl]: Sorry, mixed my local branch to this change, removing it now.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/private-cluster/variables.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/private-cluster/variables.tf:379]: @ericyz FYI I switched it back since next release is breaking https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1421/commits/731e5ec00b7e88ad5651a355f220ee98bdce3988

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@ericyz
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

**Comment by gamepop** [comment]
Does this change Global master access default to True
